network/debinterfaces/activate.go

```diff
@@ -79,8 +79,18 @@ fi
 ${DRYRUN} write_content %[3]q
 ${DRYRUN} ifdown --interfaces=%[1]q %[7]s
 ${DRYRUN} sleep %[4]d
-${DRYRUN} ifup --interfaces=%[3]q -a
-${DRYRUN} mv %[3]q %[1]q
+${DRYRUN} cp %[3]q %[1]q
+# we want to have full control over what happens next
+set +e
+${DRYRUN} ifup --interfaces=%[1]q -a
+RESULT=$?
+if [ ${RESULT} != 0 ]; then
+    echo "Bringing up bridged interfaces failed, see system logs and %[3]q" >&2
+    ${DRYRUN} ifdown --interfaces=%[1]q %[7]s
+    ${DRYRUN} cp %[2]q %[1]q
+    ${DRYRUN} ifup --interfaces=%[1]q -a
+    exit ${RESULT}
+fi
 `,
 		params.Filename,
 		params.Filename+".backup",
```

Comment:
let's change this to use a timestamped backup, so that we can keep multiple versions. Since we'd like to have both the '/e/n/i that we started with' and the 'last known good value'.